src/librustc_resolve/lib.rs

```diff
@@ -3229,9 +3226,12 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
             }
         }
 
+        // As a loose rule to avoid obviously incorrect suggestions, clamp the
+        // maximum edit distance we will accept for a suggestion to one third of
+        // the typo'd name's length.
+        let max_distance = std::cmp::max(name.len(), 3) / 3;
```

Comment:
Nah, if it were `min` instead of `max`, it would be, but for example, if `name.len()` was `9`, `max(9,3)` is `9`, so max_distance would be `9 / 3` or `3`.